sdk/core/azure-core/src/main/java/com/azure/core/http/rest/SwaggerMethodParser.java

```diff
@@ -201,11 +167,17 @@
                     final BodyParam bodyParamAnnotation = (BodyParam) annotation;
                     bodyContentMethodParameterIndex = parameterIndex;
                     bodyContentType = bodyParamAnnotation.value();
-                    bodyJavaType = swaggerMethod.getGenericParameterTypes()[parameterIndex];
+
+                    // If the body Content-Type is "application/x-www-form-urlencoded" the real body type is a String.
+                    bodyJavaType = (ContentType.APPLICATION_X_WWW_FORM_URLENCODED.equals(bodyContentType))
+                        ? String.class : swaggerMethod.getGenericParameterTypes()[parameterIndex];
                 } else if (annotationType.equals(FormParam.class)) {
                     final FormParam formParamAnnotation = (FormParam) annotation;
-                    formSubstitutions.add(new Substitution(formParamAnnotation.value(), parameterIndex,
```

Comment:
>formSubstitutions [](start = 20, length = 17)

when will autorest generate @FormParam vs @BodyParam for APPLICATION_X_WWW_FORM_URLENCODED

Reply:
@jianghaolu, @srnagar ?

Reply:
Currently AutoRest doesn't support generating `FormParam` due to lack of support in azure-core. You tell us what to generate for the following scenarios and we can generate as desired:

1. Upload file through multipart form-data. See [swagger](https://github.com/Azure/autorest.testserver/blob/master/swagger/body-formdata.json#L42). Consumes `multipart/form-data`. I expect this to be a `FormParam`.
2. Upload file through body. See [swagger](https://github.com/Azure/autorest.testserver/blob/master/swagger/body-formdata.json#L79). Consumes `application/octet-stream`. I expect this to be a `BodyParam`.
3. Upload file through form url encoded form-data. See [swagger](https://github.com/Azure/autorest.testserver/blob/master/swagger/body-formdata-urlencoded.json#L46). Consumes `application/x-www-form-urlencoded`. I prefer this to be a `FormParam` but then how do we differentiate with multi-part? I'm okay with either on this.

Reply:
I would use `FormPart` for `multipart/form-data` given that we've already tied `FormParam` to URL encoded